lib/job-iteration/active_record_cursor.rb

```diff
@@ -18,12 +18,8 @@ def initialize
       end
     end
 
-    def initialize(relation, columns = nil, position = nil)
-      @columns = if columns
-        Array(columns)
-      else
-        Array(relation.primary_key).map { |pk| "#{relation.table_name}.#{pk}" }
-      end
+    def initialize(relation, columns, position = nil)
+      @columns = columns
       self.position = Array.wrap(position)
       raise ArgumentError, "Must specify at least one column" if columns.empty?
       if relation.joins_values.present? && !@columns.all? { |column| column.to_s.include?(".") }
```

Comment:
is this still necessary, now that it's an arel attribute being passed in instead of a string?

Reply:
This logic was left untouched because on the presence of join conditions, the ORDER by columns can become ambiguous and fails the queries.
The `.all?` clause was a constraint to prevent this failure mode.

Also FWIW, the changes in this PR were reverted in #483 (v1.5.1). There's more context about the bug it introduced in #483.